homeassistant/components/tautulli/sensor.py

```diff
@@ -59,90 +57,38 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     verify_ssl = config.get(CONF_VERIFY_SSL)
 
     session = async_get_clientsession(hass, verify_ssl)
-    tautulli = TautulliData(
-        PyTautulli(
-            api_token=api_key,
-            hostname=host,
-            session=session,
-            verify_ssl=verify_ssl,
-            port=port,
-            ssl=use_ssl,
-            base_api_path=path,
-        )
+    api_client = PyTautulli(
+        api_token=api_key,
+        hostname=host,
+        session=session,
+        verify_ssl=verify_ssl,
+        port=port,
+        ssl=use_ssl,
+        base_api_path=path,
     )
 
-    await tautulli.async_update()
-    if not tautulli.activity or not tautulli.home_stats or not tautulli.users:
+    coordinator = TautulliDataUpdateCoordinator(hass=hass, api_client=api_client)
+
+    await coordinator.async_refresh()
+    if not coordinator.activity or not coordinator.home_stats or not coordinator.users:
```

Comment:
coordinator exposes a last_update_success property, which would be False if UpdateFailed took place

Reply:
Which would be great :+1 But It can be successful, but not return this data..

Reply:
I thought that might be the case all good. might be good to check it still in addition to your other checks as there are several reasons it might fail from coordinator

Reply:
Added guard for UpdateFailed